src/math_parser_diag.cpp

```diff
@@ -443,6 +569,136 @@ std::function<void( dialogue &, double )> spell_exp_ass( char scope,
     };
 }
 
+std::function<double( dialogue & )> spell_level_eval( char scope,
+        std::vector<diag_value> const &params, diag_kwargs const &/* kwargs */ )
+{
+    return[beta = is_beta( scope ), spell_value = params[0]]( dialogue const & d ) {
+        const spell_id spell( spell_value.str( d ) );
+        if( spell == spell_id::NULL_ID() ) {
+            return d.actor( beta )->get_highest_spell_level();
+        } else {
+            return d.actor( beta )->get_spell_level( spell );
+        }
+    };
+}
+
+std::function<void( dialogue &, double )> spell_level_ass( char scope,
+        std::vector<diag_value> const &params, diag_kwargs const &/* kwargs */ )
+{
+    return[beta = is_beta( scope ), spell_value = params[0]]( dialogue const & d, double val ) {
+        const spell_id spell( spell_value.str( d ) );
+        if( spell == spell_id::NULL_ID() ) {
+            throw std::invalid_argument( string_format( "Can't set spell level of %s", spell.str() ) );
```

Comment:
`throw` inside the lambda will just crash. This should be a `debugmsg`.

Reply:
replaced to `debugmsg`